midterm-2023-11-06/code/5090-teodor-inovski/script.ts

```diff
@@ -0,0 +1,209 @@
+document.addEventListener("DOMContentLoaded", siteCode)
+
+//band-table
+//band-data
+
+
+interface Album{
+    name: string;
+    year: number;
+}
+
+interface Band {
+    id: number;
+    name: string;
+    genre: string;
+    formed: number;
+    location: string;
+    members: string[];
+    albums: Album[];
+}
+
+type BandSorter = (first: Band, second: Band) => number;
+
+let bands: Band[]= [];
+
+
+async function siteCode(){
+   
+
+    const data = await loadData();
+    
+    bands=data.metalBands;
+    
+    fillGenres(bands)
+
+    displayBands(bands);
+
+    const nameSort= document.getElementById("sort-name")
+    nameSort?.addEventListener("click", sortByName);
+
+    const idSort= document.getElementById("sort-id")
+    idSort?.addEventListener("click", sortById);
+
+    const formedSort= document.getElementById("sort-formed")
+    formedSort?.addEventListener("click", sortByFormed);
+
+    const locationSort= document.getElementById("sort-location")
+    locationSort?.addEventListener("click", sortByLocation);
+
+    const albumSort= document.getElementById("sort-albums")
+    albumSort?.addEventListener("click", sortByAlbums);
+
+    const memberSort= document.getElementById("sort-member")
+    memberSort?.addEventListener("click",sortByMembers)
+    
+    const applyFilterButton = document.getElementById("apply-filter")!;
+    applyFilterButton.addEventListener("click", applyFilter)
+
+}
+
+const nameSorter: BandSorter=(first,second)=> first.name.localeCompare(second.name)
```

Comment:
Functional sort implementation, but they don't take the nuances of e.g. album and members sorting. Also no computed properties (country) sorting
+10